### PR TITLE
Wire up publish all

### PR DESCRIPTION
**Pull request by jeroach**
[US108297](https://rally1.rallydev.com/#/detail/userstory/318899091508?fdp=true): [UI] Hook Publish All for  all activities - Currently supported Assignments and Discussions

This is just the logic - there is a follow up story to handle the confirmation dialog/error messages/etc.

### CODE REVIEW

File: test/d2l-quick-eval/d2l-quick-eval-activities.js [hunk 1]
```diff
@@ -100,7 +112,23 @@ suite('d2l-quick-eval-activities', function() {
 		function checkData() {
 			if (act._data.length) {
 				assert.equal(act._data.length, expectedData.length);
-				assert.deepEqual(act._data, expectedData);
+
+				for (let i = 0; i < act._data.length; i++) {
```
Comment (jeroach) [inline test/d2l-quick-eval/d2l-quick-eval-activities.js]: Had a lot of issues getting `deepEqual` to work - mainly due to the `Action` object being returned would have to match exactly, and I was struggling with it.

If anyone has a suggestion here, I'm all ears :)

File: demo/index.html [hunk 2]
```diff
@@ -27,6 +27,8 @@
 				</custom-style>
 			`;
 			document.body.appendChild($_documentContainer.content);
+
+			window.D2L.Siren.WhitelistBehavior._testMode(true);
```
Comment (jeroach) [inline demo/index.html]: Added to remove the console error messages in the demo due to the whitelist code

File: demo/d2l-quick-eval/interceptor.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ function parsePathNoLeadingSlash(input) {
 }
 
 function startMockServer(mappings, debug, delay) {
+	window.D2L.Siren.WhitelistBehavior._testMode(true);
```
Comment (jeroach) [inline demo/d2l-quick-eval/interceptor.js]: Added to remove the console error messages in the demo due to the whitelist code

File: components/d2l-quick-eval/activity-card/d2l-quick-eval-activity-card.js [hunk 4]
```diff
@@ -154,6 +162,10 @@ class D2LQuickEvalActivityCard extends QuickEvalLocalize(PolymerElement) {
 		this._focused = !this._focused;
 	}
 
+	_publishAll() {
+		this.performSirenAction(this.publishAll);
```
Comment (jeroach) [inline components/d2l-quick-eval/activity-card/d2l-quick-eval-activity-card.js]: SirenAction for the win :)

File: test/d2l-quick-eval/d2l-quick-eval-activities.js (hunk 1 above)
Comment (CarolBusman) [reply test/d2l-quick-eval/d2l-quick-eval-activities.js]: Does the Action object have "private" or computed fields? I think I had the same problem earlier for the same reason, and ended up writing an equality checker that ignored anything starting with `_`.

File: components/d2l-quick-eval/activity-card/d2l-quick-eval-activity-card.js [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
 import 'd2l-icons/d2l-icon.js';
 import 'd2l-icons/tier3-icons.js';
 import 'd2l-polymer-behaviors/d2l-visible-on-ancestor-behavior.js';
+import 'd2l-polymer-siren-behaviors/store/siren-action-behavior.js';
 import './d2l-quick-eval-activity-card-items.js';
 import './d2l-quick-eval-activity-card-unread-submissions.js';
 
-class D2LQuickEvalActivityCard extends QuickEvalLocalize(PolymerElement) {
+class D2LQuickEvalActivityCard extends mixinBehaviors(
+	[D2L.PolymerBehaviors.Siren.SirenActionBehavior],
```
Comment (CarolBusman) [inline components/d2l-quick-eval/activity-card/d2l-quick-eval-activity-card.js]: I'm wondering if we actually want to bring in Siren stuff at this level. Does it make sense to keep the activity card dumb/"display only" and just fire an event for the controller to handle when a button is clicked on?

File: test/d2l-quick-eval/d2l-quick-eval-activities.js (hunk 1 above)
Comment (jeroach) [reply test/d2l-quick-eval/d2l-quick-eval-activities.js]: Yeah it does have computed fields. There is also the object type itself - my `expectedData` JSON is just JSON, where as the returned `act._data...publishAll` is an `Action` object

File: components/d2l-quick-eval/behaviors/d2l-quick-eval-siren-helper-behavior.js [hunk 6]
```diff
@@ -82,13 +82,20 @@ D2L.PolymerBehaviors.QuickEval.D2LQuickEvalSirenHelperBehaviorImpl = {
 			.then(function(e) {
 				if (e && e.entity && e.entity.properties) {
 					const p = e.entity.properties;
+
+					let publishAll = {};
```
[the remark this answers is not in the document]
Comment (jeroach) [reply components/d2l-quick-eval/behaviors/d2l-quick-eval-siren-helper-behavior.js]: Good point! And it doesn't need to be `{}` - that was just the default value I chose. Will update.

File: components/d2l-quick-eval/activity-card/d2l-quick-eval-activity-card.js (hunk 5 above)
Comment (jeroach) [reply components/d2l-quick-eval/activity-card/d2l-quick-eval-activity-card.js]: Another good point! I'll look into moving this out.

**Comment by jeroach** [comment]
Need to fix some console errors around the event being null

File: components/d2l-quick-eval/d2l-quick-eval-activities.js [hunk 7]
```diff
@@ -20,7 +21,11 @@ import './activities-list/d2l-quick-eval-activities-list.js';
  */
 
 class D2LQuickEvalActivities extends mixinBehaviors(
-	[D2L.PolymerBehaviors.QuickEval.D2LQuickEvalSirenHelperBehavior, D2L.PolymerBehaviors.QuickEval.D2LHMFilterBehaviour, D2L.PolymerBehaviors.QuickEval.D2LHMSearchBehaviour],
+	[	D2L.PolymerBehaviors.QuickEval.D2LQuickEvalSirenHelperBehavior,
+		D2L.PolymerBehaviors.QuickEval.D2LHMFilterBehaviour,
+		D2L.PolymerBehaviors.QuickEval.D2LHMSearchBehaviour,
+		D2L.PolymerBehaviors.Siren.SirenActionBehavior
```
Comment (CarolBusman) [inline components/d2l-quick-eval/d2l-quick-eval-activities.js]: Do we need to import this? I think `D2LQuickEvalSirenHelperBehavior` comes with `SirenActionBehavior`?

**Pull request merged**